nursery/coin_select/src/lib.rs

```diff
@@ -1,33 +1,61 @@
 #![no_std]
+// #![warn(missing_docs)]
+#![doc = include_str!("../README.md")]
+#![deny(unsafe_code)]
 
-#[cfg(feature = "std")]
-extern crate std;
 
+#[allow(unused)]
 #[macro_use]
 extern crate alloc;
-extern crate bdk_chain;
 
-use alloc::vec::Vec;
-use bdk_chain::{
-    bitcoin,
-    collections::{BTreeSet, HashMap},
-};
-use bitcoin::{LockTime, Transaction, TxOut};
-use core::fmt::{Debug, Display};
+#[cfg(feature = "std")]
+#[allow(unused)]
+#[macro_use]
+extern crate std;
+
 
 mod coin_selector;
+pub mod float;
 pub use coin_selector::*;
 
 mod bnb;
 pub use bnb::*;
 
-/// Txin "base" fields include `outpoint` (32+4) and `nSequence` (4). This does not include
-/// `scriptSigLen` or `scriptSig`.
-pub const TXIN_BASE_WEIGHT: u32 = (32 + 4 + 4) * 4;
+pub mod metrics;
+
+mod feerate;
+pub use feerate::*;
+pub mod change_policy;
+
+/// Txin "base" fields include `outpoint` (32+4) and `nSequence` (4) and 1 byte for the scriptSig
+/// length.
+pub const TXIN_BASE_WEIGHT: u32 = (32 + 4 + 4 + 1) * 4;
+
+/// The weight of a TXOUT without the `scriptPubkey` (and script pubkey length field).
+/// Just the weight of the value field.
+pub const TXOUT_BASE_WEIGHT: u32 = 4 * core::mem::size_of::<u64>() as u32; // just the value
```

Comment:
nit: Technically it's an `i64`, but the size is the same anyways.

```suggestion
pub const TXOUT_BASE_WEIGHT: u32 = 4 * core::mem::size_of::<i64>() as u32; // just the value
```

Reply:
I actually didn't know that bitcoin core treated these at `i64` internally. I think from a consensus perspective either is correct? i.e. if we changed the definition to be `u64` the protocol doesn't change at all.